Sources/Orbit/Support/Components/TimelineItem.swift

```diff
@@ -0,0 +1,440 @@
+        switch style {
+            case .future:
+                Circle()
+                    .strokeBorder(style.color, lineWidth: 2)
+                    .background(Circle().fill(Color.whiteNormal))
+                    .frame(width: .small * sizeCategory.ratio, height: .small * sizeCategory.ratio)
+            case .current(.info):
+                ZStack {
+                    Circle()
+                        .frame(
+                            width: (animationLoopTrigger ? .medium : .xMedium) * sizeCategory.ratio,
+                            height: (animationLoopTrigger ? .medium : .xMedium) * sizeCategory.ratio
+                        )
+                        .animation(Animation.easeInOut.repeatForever().speed(animationSpeed))
+                        .foregroundColor(style.color.opacity(0.1))
+
+                    Circle()
+                        .strokeBorder(style.color, lineWidth: 2)
+                        .background(Circle().fill(Color.whiteNormal))
+                        .frame(width: .small * sizeCategory.ratio, height: .small * sizeCategory.ratio)
+
+                    Circle()
+                        .frame(width: .xxSmall * sizeCategory.ratio, height: .xxSmall * sizeCategory.ratio)
+                        .scaleEffect(animationLoopTrigger ? .init(width: 0.5, height: 0.5) : .init(width: 1, height: 1))
+                        .foregroundColor(animationLoopTrigger ? Color.clear : style.color)
+                        .animation(Animation.easeInOut.repeatForever().speed(animationSpeed))
+
+                }
+        case .current(.warning), .current(.critical), .current(.success), .past:
+                Icon(style.iconSymbol, size: .small, color: style.color)
+                    .background(Circle().fill(Color.whiteNormal).padding(2))
+        }
+    }
+}
+
+// MARK: - Inits
+
+public extension TimelineItem {
+
+    /// Creates Orbit TimelineItem component with text details.
+    init(
+        _ label: String = "",
+        sublabel: String = "",
+        style: TimelineItemStyle = .future,
+        description: String = "",
+        @ViewBuilder footer: () -> Footer
+    ) {
+
+        self.label = label
+        self.sublabel = sublabel
+        self.style = style
+        self.description = description
+        self.footer = footer()
+    }
+}
+
+public extension TimelineItem where Footer == EmptyView {
+
+    /// Creates Orbit TimelineItem component with text details.
+    init(
+        _ label: String,
+        sublabel: String = "",
+        style: TimelineItemStyle = .future,
+        description: String = ""
+    ) {
+
+        self.label = label
+        self.sublabel = sublabel
+        self.style = style
+        self.description = description
+        self.footer = EmptyView()
+    }
+}
+
+// MARK: - Types
+
+public enum TimelineItemStyle: Equatable {
+    case past
+    case current(Status)
+    case future
+
+    public static let indicatorDiameter: CGFloat = Icon.Size.large.value
```

Comment:
This probably belongs more to the TimelineItem component

Reply:
This property is used both in `Timeline` and `TimelineItem`, at the same time it cannot be a static property of `TimelineItem` because it has a generic type. Doesn't it make sense to keep it in here?